internal/api/mfa.go

```diff
@@ -209,11 +208,10 @@ func (a *API) VerifyFactor(w http.ResponseWriter, r *http.Request) error {
 		return internalServerError(InvalidFactorOwnerErrorMessage)
 	}
 
-	challenge, err := models.FindChallengeByChallengeID(a.db, params.ChallengeID)
-	if err != nil {
-		if models.IsNotFoundError(err) {
-			return notFoundError(err.Error())
-		}
+	challenge, err := models.FindChallengeByID(a.db, params.ChallengeID)
+	if err != nil && models.IsNotFoundError(err) {
+		return notFoundError(err.Error())
+	} else if err != nil {
```

Comment:
unrelated to this particular line but i noticed that in the transaction below, we are just running a single delete query 

we can probably remove the transaction and just delete the challenge

Reply:
That makes sense, thanks for the catch!